hadoop-ozone/fault-injection-test/mini-chaos-tests/src/test/java/org/apache/hadoop/ozone/MiniOzoneChaosCluster.java

```diff
@@ -406,19 +280,71 @@ public MiniOzoneChaosCluster build() throws IOException {
       final List<HddsDatanodeService> hddsDatanodes = createHddsDatanodes(
           scm, null);
 
-      MiniOzoneChaosCluster cluster;
-      if (failureService == FailureService.DATANODE) {
-        cluster = new MiniOzoneDatanodeChaosCluster(conf, omList, scm,
-            hddsDatanodes, omServiceId);
-      } else {
-        cluster = new MiniOzoneOMChaosCluster(conf, omList, scm,
-            hddsDatanodes, omServiceId);
-      }
+      MiniOzoneChaosCluster cluster =
+          new MiniOzoneChaosCluster(conf, omList, scm, hddsDatanodes,
+              omServiceId, clazzes);
 
       if (startDataNodes) {
         cluster.startHddsDatanodes();
       }
       return cluster;
     }
   }
+
+  // OzoneManager specifc
+  public List<OzoneManager> omToFail() {
+    int numNodesToFail = FailureManager.getNumberOfOmToFail();
+    if (failedOmSet.size() >= numOzoneManagers/2) {
+      return Collections.emptyList();
+    }
+
+    int numOms = getOzoneManagersList().size();
+    List<OzoneManager> oms = new ArrayList<>(numNodesToFail);
+    for (int i = 0; i < numNodesToFail; i++) {
+      int failedNodeIndex = FailureManager.getBoundedRandomIndex(numOms);
+      oms.add(getOzoneManager(failedNodeIndex));
```

Comment:
Coverted into hashset.

Reply:
There are two concerns with adding same node multiple times:

1. trying to fail the same nodes multiple times
2. not failing the required number of nodes

I think using a set addresses the first item, but not the second one.  This could be fixed by generating a random combination of the required size instead of generating multiple independent random indexes.

I guess that's OK to address in a followup, because this was broken previously, too.